src/Runtime/Distributions/Automata/Automaton.cs

```diff
@@ -1011,45 +1030,49 @@ public void SetGroup(int group)
         /// </summary>
         /// <returns>The logarithm of the normalizer.</returns>
         /// <remarks>Returns <see cref="double.PositiveInfinity"/> if the sum diverges.</remarks>
-        public double GetLogNormalizer() => this.DoGetLogNormalizer(false);
+        public double GetLogNormalizer() => DoGetLogNormalizer(false, out var _);
 
         /// <summary>
         /// Normalizes the automaton so that the sum of its values over all possible sequences equals to one
         /// and returns the logarithm of the normalizer.
         /// </summary>
+        /// <param name="result">Will be set to the normalized automaton if the normalization was successful, or to the current automaton otherwise.</param>
         /// <returns>The logarithm of the normalizer.</returns>
         /// <exception cref="InvalidOperationException">Thrown if the automaton cannot be normalized (i.e. if the normalizer is zero or positive infinity).</exception>
         /// <remarks>The only automaton which cannot be normalized, but has a finite normalizer, is zero.</remarks>
-        public double NormalizeValues()
+        public double NormalizeValues(out TThis result)
```

Comment:
Since this is a shorthand for TryNormalizeValues, it should just return the normalized automaton.